aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/util/CognitoAuthExceptionConverter.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.util;
+
+import com.amplifyframework.auth.AuthException;
+
+import com.amazonaws.services.cognitoidentityprovider.model.AliasExistsException;
+import com.amazonaws.services.cognitoidentityprovider.model.CodeDeliveryFailureException;
+import com.amazonaws.services.cognitoidentityprovider.model.CodeMismatchException;
+import com.amazonaws.services.cognitoidentityprovider.model.ExpiredCodeException;
+import com.amazonaws.services.cognitoidentityprovider.model.InvalidParameterException;
+import com.amazonaws.services.cognitoidentityprovider.model.InvalidPasswordException;
+import com.amazonaws.services.cognitoidentityprovider.model.LimitExceededException;
+import com.amazonaws.services.cognitoidentityprovider.model.PasswordResetRequiredException;
+import com.amazonaws.services.cognitoidentityprovider.model.ResourceNotFoundException;
+import com.amazonaws.services.cognitoidentityprovider.model.TooManyFailedAttemptsException;
+import com.amazonaws.services.cognitoidentityprovider.model.UserNotConfirmedException;
+import com.amazonaws.services.cognitoidentityprovider.model.UserNotFoundException;
+import com.amazonaws.services.cognitoidentityprovider.model.UsernameExistsException;
+
+/**
+ * Convert AWS Cognito Exceptions to AuthExceptions.
+ */
+public final class CognitoAuthExceptionConverter {
+
+    /**
+     * Dis-allows instantiation of this class.
+     */
+    private CognitoAuthExceptionConverter() {}
+
+    /**
+     * Lookup method to convert AWS Cognito Exception to AuthException.
+     * @param error Exception thrown by AWSMobileClient
+     * @param fallbackMessage Fallback message to inform failure
+     * @return AuthException
+     */
+    public static AuthException lookup(Exception error, String fallbackMessage) {
```

Comment:
Prefix with `@NonNull` to let developers know that this operation always returns a non-null exception.
The parameters should also have `@NonNull` and/or `@Nullable` tags.

The above practice should be used for every publicly exposed methods throughout the repo :)